Cabal/Distribution/Simple/Build/Macros.hs

```diff
@@ -99,10 +99,10 @@ generateMacros prefix name version =
 -- | Generate the @CURRENT_PACKAGE_KEY@ definition for the package key
 --   of the current package, if supported by the compiler.
 --   NB: this only makes sense for definite packages.
-generatePackageKeyMacro :: LocalBuildInfo -> String
-generatePackageKeyMacro lbi
+generateComponentInstanceIdMacro :: LocalBuildInfo -> String
+generateComponentInstanceIdMacro lbi
   | packageKeySupported (compiler lbi) =
-      "#define CURRENT_PACKAGE_KEY \"" ++ display (localPackageKey lbi) ++ "\"\n\n"
+      "#define CURRENT_PACKAGE_KEY \"" ++ display (localComponentInstanceId lbi) ++ "\"\n\n"
```

Comment:
We may want to deprecate `CURRENT_PACKAGE_KEY` and introduce a `CURRENT_COMPONENT_INSTANCE_ID` macro.